WordPress/gutenberg
# List View: Update draggable chip to resemble the list view item when dragged

## What?


Part of https://github.com/WordPress/gutenberg/issues/49563

This is an exploratory PR to look into customising the draggable chip that's used when dragging blocks within the list view. The goal is to make it feel like the user is dragging the real list item, rather than using an abstraction for the drag chip.

Note: 🚧 🎨 this is just an experiment at the moment, it'll need design feedback on what the chip should ideally look like. The opacity is added so that the drop indicator line isn't too obscured.

## Why?


There are two main goals here:

1. Make the chip more closely resemble the list view items, so that moving items around feels a little closer to "you've picked up that row you're dragging"
2. Reduce how the chip obscures the list view drop indicator line.

To elaborate on the second point above: while working on dragging to all levels of the block hierarchy (https://github.com/WordPress/gutenberg/pull/49742) it's become clear that when dragging to higher levels of the hierarchy, we need the chip not to obscure the drop indicator line, otherwise it's hard for users to see where they'll be dropping the dragged block.

### Not covered

* The mouse cursor icon is not addressed in this PR

## How?


* Update the `BlockDraggable` component to accept an `__experimentalDragComponent` and pass it down to the `Draggable` component.
* Also allow the component passed to `Draggable` to be overridden, and for `BlockDraggable` to accept an `elementId`
* In the List View, set the draggable component to `null`, and pass `BlockDraggable` the id of the list view row — this tells `Draggable` to make a clone of the list view item for dragging
* Add some styling for the cloned version of the list view item, so that it more closely resembles the list view item while dragged.

## Testing Instructions





1. Open a post or page with a bunch of blocks in it.
2. Try dragging blocks around and see how the chip looks with this PR applied.
3. Try adding a long heading, and add an `id` attribute in the Advanced panel. It should display the heading title and id attribute in a way that's pretty similar to the non-dragged version.

### Testing Instructions for Keyboard


## Screenshots or screencast 

| Before | After |
| --- | --- |
| <img width="469" alt="image" src="https://user-images.githubusercontent.com/14988353/231952844-adcd5094-531f-4e78-aa50-442a97ca6512.png"> | <img width="548" alt="image" src="https://user-images.githubusercontent.com/14988353/232691056-849fc8e5-1140-4c99-a795-2ec86889b20e.png"> |

Screengrab:

https://github.com/WordPress/gutenberg/assets/14988353/0a2d9f23-8819-4ac4-a6c3-8f3c24d50de1
